### PR TITLE
Exception handling in query execution [Redo]

### PR DESCRIPTION
**Pull request by ChTimTsubasa**
This PR redo the #1048 

Some differences include:
1. This PR relies on p_status instead of passing error_message as argument
2. This PR eliminate error_message passing as an argument to Traffic Cop. Further member variable refactors would be done in a later PR.
3. This PR does not address the constraint violation[https://github.com/cmu-db/peloton/pull/1048/commits/a78a7a66936ff1b17552c8af91df5e09373edaa1#diff-06c50db2ff0eaf7f264747f748e905c0R347] since this is beyond of the scope of my design decision and should be done by someone familiar with that. That person should decide whether he or she should rely on return value or just throw an exception.

### CODE REVIEW

**Comment by ChTimTsubasa** [comment]
#1062 is addressed.

File: src/executor/plan_executor.cpp [hunk 1]
```diff
@@ -64,19 +63,19 @@ static void CompileAndExecutePlan(
...
-            auto str = column_val.IsNull() ? "" : column_val.ToString();
-            LOG_TRACE("column content: [%s]", str.c_str());
-            values.push_back(std::move(str));
-          }
-        }
-        on_complete(result, std::move(values));
-      };
+  auto on_query_result = [&on_complete,
+                          &consumer](executor::ExecutionResult result) {
+    std::vector<ResultValue> values;
```
Comment (pmenon) [inline src/executor/plan_executor.cpp]: There isn't any logic change here, but are you using `clang-format-3.6` for formatting?

File: src/executor/plan_executor.cpp [hunk 2]
```diff
@@ -101,6 +100,7 @@ static void InterpretPlan(
   status = executor_tree->Init();
   if (status != true) {
     result.m_result = ResultType::FAILURE;
+    result.m_error_message = "Executor tree init failed";
```
Comment (pmenon) [inline src/executor/plan_executor.cpp]: Can you make this a little more descriptive? Something like "_Failed initialization of query execution tree_"?

File: src/executor/plan_executor.cpp [hunk 3]
```diff
@@ -146,10 +146,19 @@ void PlanExecutor::ExecutePlan(
...
+    if (codegen_enabled && codegen::QueryCompiler::IsSupported(*plan)) {
+      CompileAndExecutePlan(plan, txn, params, on_complete);
+    } else {
+      InterpretPlan(plan, txn, params, result_format, on_complete);
+    }
+  } catch (Exception &e) {
+    ExecutionResult result;
+    result.m_result = ResultType::FAILURE;
+    result.m_error_message = e.what();
+    LOG_ERROR("error thrown in Execution: %s", result.m_error_message.c_str());
```
Comment (pmenon) [inline src/executor/plan_executor.cpp]: Let's be consistent about starting sentences with capitals.

File: src/planner/insert_plan.cpp [hunk 4]
```diff
@@ -37,7 +37,9 @@ InsertPlan::InsertPlan(storage::DataTable *table,
     for (uint32_t tuple_idx = 0; tuple_idx < insert_values->size();
          tuple_idx++) {
       auto &values = (*insert_values)[tuple_idx];
-      PL_ASSERT(values.size() <= schema->GetColumnCount());
+      if (values.size() > schema->GetColumnCount()) {
```
Comment (pmenon) [inline src/planner/insert_plan.cpp]: Remove these for now and they'll be addressed in #1063

**Pull request merged**